services/gateway/tests/unit/registerHttpApi.test.js

```diff
@@ -0,0 +1,192 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	convertType,
+	mapParam,
+	getMethodName,
+	dropOneSlashAtBeginning,
+	curlyBracketsToColon,
+} = require('../../shared/registerHttpApi');
+const { sourceForMapParam } = require('../constants/registerApi');
+
+describe('Test convertType method', () => {
+	it('should return number when called to convert string to number', async () => {
+		const response = convertType('123', 'number');
+		expect(typeof response).toEqual('number');
+		expect(response).toEqual(123);
+	});
+
+	it('should return string when called to convert number to string', async () => {
+		const response = convertType(123, 'string');
+		expect(typeof response).toEqual('string');
+		expect(response).toEqual('123');
+	});
+
+	it('should return string when called to convert array to string', async () => {
+		const response = convertType(new Array(...[1, 2, 3]), 'string');
+		expect(typeof response).toEqual('string');
+		expect(response).toEqual('1,2,3');
+	});
```

Comment:
What happens when we input `[1, 2, 3]` instead of instantiating an `Array` object?
The mentioned scenario must be handled too.

Reply:
Added the test case for the mentioned scenario.